### PR TITLE
Fix point handling in path_to_geos.

### PR DESCRIPTION
**Pull request by pelson**
Fixes a case that came from a matplotlib path. Ping @marqh.

### CODE REVIEW

**Comment by marqh** [comment]
this meets my requirements, fixing a contouring problem which was giving me headaches

**Comment by marqh** [comment]
the point for a degenerate polygon is a neat fix, i'm happy with it.

File: lib/cartopy/tests/mpl/test_shapely_to_mpl.py [hunk 1]
```diff
@@ -81,15 +81,6 @@ def test_polygon_interiors():
     ax.add_collection(collection)
 
 
-def test_null_geometry():
-    pth = Path([[358.27203369, 3.56399965],
-                [358.27203369, 3.56399965],
-                [358.27203369, 3.56399965]])
-    geoms = cpatch.path_to_geos(pth)
-
-    assert len(geoms) == 0
```
Comment (marqh) [inline lib/cartopy/tests/mpl/test_shapely_to_mpl.py]: not sure why this is removed, wouldn't it just return len(geoms) = 1 now?

File: lib/cartopy/mpl/patch.py [hunk 2]
```diff
@@ -208,6 +212,8 @@ def path_to_geos(path, force_ccw=False):
                 isinstance(geom, Polygon) and
                 collection[-1][0].contains(geom.exterior)):
             collection[-1][1].append(geom.exterior)
+        elif isinstance(geom, Point):
+            other_result_geoms.append(geom)
```
Comment (marqh) [inline lib/cartopy/mpl/patch.py]: is this other_results_geoms collection just to enable the points not to be filtered out below?

File: lib/cartopy/tests/mpl/test_shapely_to_mpl.py (hunk 1 above)
Comment (pelson) [inline lib/cartopy/tests/mpl/test_shapely_to_mpl.py]: Yep. But the test is redundant, and in the wrong place.

File: lib/cartopy/mpl/patch.py (hunk 2 above)
Comment (pelson) [inline lib/cartopy/mpl/patch.py]: Yep. It's not perfect, as we lose some of the ordering, but it fits the requirements for now.

**Comment by marqh** [comment]
this looks good to me; fixes my bug neatly.  I don't think there's much to add from a refactor based on pelson's comments, I think the code is fine as it is

please may this be merged and staged for a 0.12.1 release?

**Pull request merged**